### PR TITLE
Allow pdo_ibm for database resources

### PR DESCRIPTION
**Pull request by kobmaki**
This enables the access to an IBM DB2 database with the driver pdo_ibm.

### CODE REVIEW

**Comment by lazyfrosch** [comment]
I took the liberty to rebase the branch and fix some whitespace issues (we don't use tabs, only spaces).

**Comment by lazyfrosch** [comment]
I guess it makes sense in terms of the base framework to allow DB2.

Supporting it and special handlings that may be required is on another argument, but we already added MSSQL and Oracle, so...

File: application/forms/Config/Resource/DbResourceForm.php [hunk 1]
```diff
@@ -54,6 +58,8 @@ public function createElements(array $formData)
             if (version_compare(Platform::getPhpVersion(), '5.4.0', '>=')) {
                 $offerSsl = true;
             }
+        } elseif ($dbChoice === 'ibm') {
+          $offerIbm = true;
```
Comment (lippserd) [inline application/forms/Config/Resource/DbResourceForm.php]: Please fix indent.

File: library/Icinga/Application/Platform.php [hunk 2]
```diff
@@ -367,6 +367,18 @@ public static function hasMysqlSupport()
...
     /**
+     * Return whether it's possible to connect to a IBM DB2 database
+     *
+     * Checks whether the ibm pdo extension has been loaded and the Zend framework adapter for IBM is available
+     *
+     * @return  bool
+     */
+    public static function hasIbmSupport()
+    {
+      return static::extensionLoaded('pdo_ibm') && static::classExists('Zend_Db_Adapter_Pdo_Ibm');
```
Comment (lippserd) [inline library/Icinga/Application/Platform.php]: Please fix indent.

**Comment by kobmaki** [comment]
I fixed the indent.

**Comment by dnsmichi** [comment]
Please remove the merge commits and cleanly rebase into one single commit.

**Comment by lazyfrosch** [comment]
Rebased into a single commit, looks like I missed those indents before :upside_down_face: 

@kobmaki have a look at:

* https://git-scm.com/book/de/v1/Git-Branching-Rebasing

**Comment by kobmaki** [comment]
I try  it. Actually I am lost in git command :-(  to do a rebase and squash with "fatal: cherry-pick failed"

**Comment by lazyfrosch** [comment]
@kobmaki I basically only did:

```
git fetch kobmaki
git fetch origin

# in my branch for your PR
git reset --hard kobmaki/Feature/DbConnection-with-pdo_ibm-to-DB2
git rebase -i origin/master
# only 2 commits should be shown in the editor
# I replaced `pick` to `f` or `fixup` in the second commit
# save, and rebase is running

git push -f
```

Send me an email with your error and I can have a look. We always try to help our contributors :wink:

**Comment by lazyfrosch** [comment]
Rebased the branch for the user...

Please review and merge @lippserd

**Pull request merged**